### PR TITLE
[IN-738] template sshd config

### PR DESCRIPTION
**Pull request by peterheja-bitrise**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [ ] ~The tool I added is stable, and __does not require frequent updates__. _Preinstalled tools on the LTS stacks
  are not updated, so if the tool requires frequent updates you should handle the installation on-demand (see: [Install Any Additional Tool - DevCenter](https://bitrise-io.github.io/devcenter/tips-and-tricks/install-additional-tools/) for more information)_~
- [ ] ~I added a version report line to [`system_report.sh`](/system_report.sh) for the new tool(s) I added~
- [ ] I have added an entry to the CHANGELOG.md with a revisionID (the current date and the number of the change e.g.: 2019_10_11_1) and a brief description of the actual change
- [ ] I have updated the revisionID in [bitrise_profile](roles/profiles/files/bitrise_profile) **BITRISE_OSX_STACK_REV_ID** with the one I just created in the CHANGELOG.md

This is minor refactor, first two points are not applicable, last two will be addressed shortly.

### CODE REVIEW

File: roles/ssh-setup/tasks/main.yml [hunk 1]
```diff
@@ -8,35 +8,13 @@
...
-    - { regexp: '^UsePAM yes', line: '', state: "absent" }
-    - { regexp: '^UsePAM no', line: 'UsePAM no', state: "present" }
-    # ChallengeResponseAuthentication
-    - { regexp: '^ChallengeResponseAuthentication yes', line: '', state: "absent" }
-    - { regexp: '^ChallengeResponseAuthentication no', line: 'ChallengeResponseAuthentication no', state: "present" }
-    # UseDNS no -> for better speed
-    - { regexp: '^UseDNS yes', line: '', state: "absent" }
-    - { regexp: '^UseDNS no', line: 'UseDNS no', state: "present" }
+  template:
+    src: templates/etc/ssh/sshd_config
```
Comment (lmesz-bitrise) [inline roles/ssh-setup/tasks/main.yml:12]: Shouldn't it have j2 extension?

Comment (bitrise-tom) [reply roles/ssh-setup/tasks/main.yml:12]: yeah, src should match the template file name

File: roles/ssh-setup/tests/test_sshd.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    sshd_config = host.file("/etc/ssh/sshd_config")
+    assert sshd_config.user == "root"
+    assert sshd_config.group == "root"
+    assert sshd_config.mode == 0o644
+
+def test_sshd_config_content(host):
+    sshd_config = host.file("/etc/ssh/sshd_config")
+    assert sshd_config.contains("Ansible managed")
+    assert sshd_config.contains("bitrise overrides")
+    assert sshd_config.contains("LogLevel")
```
Comment (lmesz-bitrise) [inline roles/ssh-setup/tests/test_sshd.py:17]: Check the value of it?

File: roles/ssh-setup/vars/main.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+home:
+  path: "/Users"
```
Comment (lmesz-bitrise) [inline roles/ssh-setup/vars/main.yaml:2]: There are built-in variable in ansible for this. Would you mind using that?

File: roles/ssh-setup/tests/test_sshd.py (hunk 2 above)
Comment (peterheja-bitrise) [reply roles/ssh-setup/tests/test_sshd.py:17]: I don't want to check values, as they are variables.
But, this ensures that the config is replaced by ours, as this is not in the original sshd_config

Comment (peterheja-bitrise) [reply roles/ssh-setup/tests/test_sshd.py:17]: Although, on second thought, maybe checking the value could be feasible.

File: roles/ssh-setup/tests/test_sshd.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+def test_sshd_config_exists(host):
+    assert host.file("/etc/ssh/sshd_config").exists
+
+def test_sshd_config_is_valid(host):
+    assert host.run("/usr/sbin/sshd -t -f /etc/ssh/sshd_config").rc == 0
```
Comment (lmesz-bitrise) [inline roles/ssh-setup/tests/test_sshd.py:5]: What do you think about using `host.run(...).succeeded `?

**Pull request merged**